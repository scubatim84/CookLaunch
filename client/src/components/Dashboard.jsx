import React from 'react';
import {Redirect} from 'react-router-dom';
import {Grid} from '@material-ui/core';
import _ from 'lodash';
import {useStylesMain} from '../Styles';
import {themeMain} from '../Theme';
import RecipeCard from './Recipes/RecipeCard';

function Dashboard(props) {
  const classes = useStylesMain(themeMain);

  if (!props.isLoggedIn) {
    return <Redirect to='/login' />;
  } else {
    return (
      <Grid container className={(classes.maxWidth, classes.root)}>
        {props.recipes.map((recipe) => {
          const formatName = _.startCase(_.toLower(recipe.name));

          return (
            <Grid
              item
              xs={6}
              sm={3}
              className={classes.recipeCard}
              key={recipe._id + recipe.dateLastChanged}
            >
              <RecipeCard
                key={recipe._id + recipe.dateLastChanged}
                id={recipe._id}
                name={formatName}
              />
            </Grid>
          );
        })}
      </Grid>
    );
  }
}

export default Dashboard;
