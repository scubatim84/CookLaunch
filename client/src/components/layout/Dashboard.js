import React, { useState, useEffect } from "react";
import { Redirect } from "react-router-dom";
import isEmpty from "is-empty";

import Typography from '@material-ui/core/Grid';

function Dashboard(props) {
  const [isLoggedIn, setLoggedIn] = useState(props.loggedIn);
  const [user, setUser] = useState(props.userData);
	
  useEffect(() => {
    setUser(props.userData)
	}, [props.userData]);

	useEffect(() => {
		setLoggedIn(props.loggedIn);
	}, [props.loggedIn]);
	
	return (
		!isLoggedIn ? <Redirect to="/login" /> : 
		<div>
			<Typography>
				Welcome {!isEmpty(user.firstName) && user.firstName}
			</Typography>
		</div>
	);
}

export default Dashboard;