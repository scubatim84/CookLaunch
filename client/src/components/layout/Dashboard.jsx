import React, {useState, useEffect} from 'react';
import {Redirect} from 'react-router-dom';
import {getUserData} from '../../actions/authActions';

import Ingredients from './Ingredients';
import Pantry from './Pantry';

import {Grid} from '@material-ui/core';

function Dashboard(props) {
  const [isLoggedIn, setLoggedIn] = useState(props.isLoggedIn);
  const [user, setUser] = useState({
    email: '',
    firstName: '',
    lastName: '',
  });

  useEffect(() => {
    setLoggedIn(props.isLoggedIn);
  }, [props.isLoggedIn]);

  useEffect(() => {
    if (isLoggedIn) {
      const getUserPayload = async () => {
        const data = await getUserData();
        const userPayload = await data.payload;

        setUser({
          email: userPayload.email,
          firstName: userPayload.firstName,
          lastName: userPayload.lastName,
        });
      };

      getUserPayload();
    }
  }, [isLoggedIn]);

  return !isLoggedIn ? (
    <Redirect to='/login' />
  ) : (
    <Grid container spacing={2}>
      <Grid item xs={12} sm={4}>
        <Ingredients
          handleLoggedIn={props.handleLoggedIn}
          isLoggedIn={props.isLoggedIn}
        />
      </Grid>
      <Grid item xs={12} sm={4}>
        <Pantry
          handleLoggedIn={props.handleLoggedIn}
          isLoggedIn={props.isLoggedIn}
        />
      </Grid>
    </Grid>
  );
}

export default Dashboard;
