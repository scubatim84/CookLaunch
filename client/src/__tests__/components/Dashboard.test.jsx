import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../../components/Dashboard';

describe('Dashboard', () => {
  it('Renders component without crashing', () => {
    shallow(<Dashboard />);
  });
});
