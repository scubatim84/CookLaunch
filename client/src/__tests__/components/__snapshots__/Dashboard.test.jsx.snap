import React from 'react';
import {shallow} from 'enzyme';
import Dashboard from '../../components/Dashboard';

describe('Dashboard', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<Dashboard />);

    expect(wrapper).toMatchSnapshot();
  });
});
